### DISCUSSION TITLE
Pagination with range query

### DISCUSSION THREAD

**Discussion by JosueRhea** [supabase/supabase#13015] [Questions] [answered]
I have some code that retrieves documents within a range. Here's the code:
```ts
export async function getDocs(
  client: SupabaseClient<Database>,
  { appId, from }: GetDocsArgs
) {
  const page = from * DOCS_LIMIT;
  return await client
    .from("doc")
    .select("*", { count: "exact" })
    .eq("appId", appId)
    .range(page, page + DOCS_LIMIT);
}
```

Suppose the doc table has `200` rows and the `DOCS_LIMIT` is set to 15. If I'm on page `from=12` and the last document on the previous page was the `195th` row, the next page should retrieve the remaining `5` rows. However, the code throws an error saying that I'm requesting 210 rows, which is more than the maximum of 200. I want to fix this issue and retrieve the remaining 5 rows. I've read through the documentation but haven't been able to find a solution. Can you please help me?

**Comment by GaryAustin1** [comment, marked as answer]
That error happens when you use `{count: "exact"}` and `range()` together.  If you drop the count it will just return less rows or 0 rows if you go over.

Note also you cannot really use range reliable without a `.order()` modifier on a UNIQUE column.  There is not default order in Postgres and you can/will get duplicate rows each time you run the `range()` request.

**Reply by JosueRhea** [reply]
thank u
